docs: Extend kubeproxy-free GSG wrt multi-dev

brb opened the pull request:
This PR:

- Adds direct routing device to `KubeProxyReplacement` in `cilium status` (denoted with `(DR)`).
- Extends the kube-proxy free GSG wrt multi-dev setup.

brb committed:
docs: Extend kubeproxy-free GSG wrt multi-dev

Signed-off-by: Martynas Pumputis <m@lambda.lt>

borkmann commented on Documentation/gettingstarted/kubeproxy-free.rst:
```diff
@@ -411,13 +411,20 @@ right at the XDP layer are not visible in tcpdump since packet taps come at a mu
-the ``global.devices`` helm option.
+LoadBalancer service or a service with externalIPs will be accessible through
+the IP addresses of native devices which either have the default route on
+the host or have Kubernetes InternalIP or ExternalIP assigned. To change
```
I would mention that these devices are auto-detected by Cilium agent. And then `To manually change [...]`

pchaigno replied:
It might be good to keep a few indication of how detection works, though I agree it should be after this paragraph.

Auto-magically is great until it stops working and then users will come asking why it doesn't behave as they expected. Having a doc. to point at would be best (unless it's already explained somewhere else?).

borkmann commented on Documentation/gettingstarted/kubeproxy-free.rst:
```diff
@@ -411,13 +411,20 @@ right at the XDP layer are not visible in tcpdump since packet taps come at a mu
+the IP addresses of native devices which either have the default route on
+the host or have Kubernetes InternalIP or ExternalIP assigned. To change
+the devices, set their names in the ``global.devices`` helm option, e.g.
+``-set global.devices={eth0,eth1,eth2}``.
```
nit: remove the `-set` (it's also `--set`), but we don't have it in other places either

borkmann replied:
Please also mention that the device names must be homogenous across nodes.

borkmann commented on Documentation/gettingstarted/kubeproxy-free.rst:
```diff
@@ -411,13 +411,20 @@ right at the XDP layer are not visible in tcpdump since packet taps come at a mu
+When multiple devices are used, only one device can be used for direct routing
+between Cilium nodes. The device can be set in the
+``global.nodePort.directRoutingDevice`` helm option. If it is not specified,
+Cilium will use a device with Kubernetes InternalIP or ExternalIP.
```
Also mention that the direct routing device is used in case of XDP as well.

And maybe a cilium status output to show where users need to look to check where it gets attached.

pchaigno commented on Documentation/gettingstarted/kubeproxy-free.rst:
```diff
@@ -411,13 +411,20 @@ right at the XDP layer are not visible in tcpdump since packet taps come at a mu
 
-NodePort Device, Port and Bind settings
+NodePort Devices, Port and Bind settings
 ***************************************
```
Doc. test is failing because the title underline also needs to be extended.

pchaigno reviewed (approved):
LGTM besides what Daniel already mentioned :point_up:

brb committed:
api,daemon,cli: Add DirectRoutingDevice to KubeProxyReplacement status

Such device is denoted with "(DR)", e.g.:

    KubeProxyReplacement:   Strict   [eth0 (DR), eth1]   [NodePort
    (SNAT, 30000-32767, XDP: DISABLED), HostPort, ExternalIPs,
    HostReachableServices (TCP, UDP), SessionAffinity]

Signed-off-by: Martynas Pumputis <m@lambda.lt>
